pinot-hadoop/src/main/java/org/apache/pinot/hadoop/job/SegmentTarPushJob.java

```diff
@@ -48,10 +54,45 @@ public void run()
       throws Exception {
     FileSystem fileSystem = FileSystem.get(_conf);
     try (ControllerRestApi controllerRestApi = getControllerRestApi()) {
-      controllerRestApi.pushSegments(fileSystem, getDataFilePaths(_segmentPattern));
+      // TODO: Deal with invalid prefixes in the future
```

Comment:
I'm not sure if segment name based solution is the correct approach for deleting segments pushed together. The logic will break for custom segment name builder. Also, it will break in the case where people are pushing to the table from multiple push jobs.

One approach that I can think of is to mark some id (e.g. batchId) to segments from offline push job and delete segments with the same batchId.

Reply:
I'm not plannign on deleting segments pushed together. I am deleting extra segments. For example,

If table_day1_0
table_day1_1
table_day1_2 are pushed today.

Table_day1_0
table_day1_1 are pushed tomorrow,

I will delete table_day1_2

Reply:
I am a little confused here -- the PR description says that you are deleting segments that are pushed or refreshed within a time range.  What exactly is an extra segment?

Reply:
discussed and resolved this